src/cregistry/entry.c

```diff
@@ -257,72 +257,53 @@ int reg_entry_delete(reg_entry* entry, reg_error* errPtr) {
                     NULL) == SQLITE_OK)
-            && (sqlite3_bind_int64(portgroups, 1, entry->id) == SQLITE_OK)) {
+            && (sqlite3_bind_int64(*pPortgroups, 1, entry->id) == SQLITE_OK)) {
         int r;
         do {
             r = sqlite3_step(ports);
             switch (r) {
                 case SQLITE_DONE:
                     if (sqlite3_changes(reg->db) > 0) {
-                        do {
-                            r = sqlite3_step(files);
-                            switch (r) {
-                                case SQLITE_DONE:
-                                    do {
-                                        r = sqlite3_step(dependencies);
-                                        switch (r) {
-                                            case SQLITE_DONE:
-                                                do {
-                                                    r = sqlite3_step(portgroups);
-                                                    switch (r) {
-                                                        case SQLITE_DONE:
-                                                            result = 1;
-                                                            break;
-                                                        case SQLITE_BUSY:
-                                                            break;
-                                                        case SQLITE_ERROR:
-                                                        default:
-                                                            reg_sqlite_error(reg->db,
-                                                                    errPtr, NULL);
-                                                            break;
-                                                    }
-                                                } while (r == SQLITE_BUSY);
-                                                break;
-                                            case SQLITE_BUSY:
-                                                break;
-                                            case SQLITE_ERROR:
-                                            default:
-                                                reg_sqlite_error(reg->db,
-                                                        errPtr, NULL);
-                                                break;
-                                        }
-                                    } while (r == SQLITE_BUSY);
-                                    break;
-                                case SQLITE_BUSY:
-                                    break;
-                                case SQLITE_ERROR:
-                                default:
-                                    reg_sqlite_error(reg->db, errPtr, NULL);
-                                    break;
-                            }
-                        } while (r == SQLITE_BUSY);
+                        size_t i;
```

Comment:
I believe we're using C99, so you can move the declaration into the initialization statement of the for loop.